ledger/acctupdates.go

```diff
@@ -1629,10 +1639,18 @@ func (au *accountUpdates) prepareCommit(dcc *deferredCommitContext) error {
 
 	// compact all the deltas - when we're trying to persist multiple rounds, we might have the same account
 	// being updated multiple times. When that happen, we can safely omit the intermediate updates.
-	dcc.compactAccountDeltas = makeCompactAccountDeltas(au.deltas[:offset], dcc.oldBase, setUpdateRound, au.baseAccounts)
-	dcc.compactResourcesDeltas = makeCompactResourceDeltas(au.deltas[:offset], dcc.oldBase, setUpdateRound, au.baseAccounts, au.baseResources)
-	dcc.compactKvDeltas = compactKvDeltas(au.kvDeltas[:offset])
-	dcc.compactCreatableDeltas = compactCreatableDeltas(au.creatableDeltas[:offset])
+	var acctDeltas []ledgercore.AccountDeltas
+	var kvDeltas []map[string]ledgercore.KvValueDelta
+	var creatableDeltas []map[basics.CreatableIndex]ledgercore.ModifiedCreatable
+	for _, sDelta := range au.deltas[:offset] {
+		acctDeltas = append(acctDeltas, sDelta.Accts)
+		kvDeltas = append(kvDeltas, sDelta.KvMods)
+		creatableDeltas = append(creatableDeltas, sDelta.Creatables)
+	}
+	dcc.compactAccountDeltas = makeCompactAccountDeltas(acctDeltas, dcc.oldBase, setUpdateRound, au.baseAccounts)
+	dcc.compactResourcesDeltas = makeCompactResourceDeltas(acctDeltas, dcc.oldBase, setUpdateRound, au.baseAccounts, au.baseResources)
+	dcc.compactKvDeltas = compactKvDeltas(kvDeltas)
+	dcc.compactCreatableDeltas = compactCreatableDeltas(creatableDeltas)
```

Comment:
I like the change that puts all the delta info in the single `deltas` field. But here it looks like it's leading to needless copying in order to call the `makeCompact*` functions with their existing signature.  Perhaps we should just give `au.deltas[:offset]` to all of them. They will have to access the specific pieces they need rather than receiving the information already narrowed down, but that seems ok to me.